### PR TITLE
Oleg merge

### PR DESCRIPTION
**Pull request by olleg12**
Added front-end  internationalization

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,7 +54,10 @@ dependencies {
     compile group: 'org.springframework', name: 'spring-webmvc', version: vers.spring
     compile group: 'org.springframework', name: 'spring-tx', version: vers.spring
     compile group: 'org.springframework', name: 'spring-orm', version: vers.spring
-    compile group: 'org.springframework', name: 'spring-test', version: vers.spring
+// https://mvnrepository.com/artifact/org.springframework/spring-test
+    compile group: 'org.springframework', name: 'spring-test', version: '4.0.2.RELEASE'
```
Comment (ViacheslavKruhlov) [inline build.gradle]: why 4.0.2?
we are using spring 4.3.5 so spring-test must be 4.3.5!

File: build.gradle [hunk 2]
```diff
@@ -83,6 +86,11 @@ dependencies {
 
     compile group: 'org.seleniumhq.selenium', name: 'htmlunit-driver', version: '2.23.2'
     compile group: 'org.seleniumhq.selenium', name: 'selenium-java', version: '3.0.1'
+
+    // https://mvnrepository.com/artifact/org.springframework.security/spring-security-test
+    compile group: 'org.springframework.security', name: 'spring-security-test', version: '4.0.1.RELEASE'
```
Comment (ViacheslavKruhlov) [inline build.gradle]: last version 4.2.1

I think, better to use last version

File: npm-debug.log [hunk 3]
```diff
@@ -0,0 +1,23 @@
+0 info it worked if it ends with ok
```
Comment (ViacheslavKruhlov) [inline npm-debug.log]: add this file to .gitignore and remove from git

git rm --cached...

File: src/main/java/ua/cv/tim/controller/PlayerController.java [hunk 4]
```diff
@@ -65,27 +69,37 @@
...
-            currentPlayer.setVillages(player.getVillages());
-            currentPlayer.setAlliance(player.getAlliance());
-            playerService.add(currentPlayer);
-
-            return new ResponseEntity<>(currentPlayer, HttpStatus.CREATED);
+        Player current_player = playerService.getById(id);
+        if (current_player != null) {
+            current_player.setRace(player.getRace());
+            current_player.setUser(player.getUser());
+            current_player.setVillages(player.getVillages());
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/controller/PlayerController.java]: naming convention - camelCase

in previous commit I changed that.

File: src/main/webapp/components/village/addVillageForm.ts [hunk 5]
```diff
@@ -21,11 +21,11 @@ export class AddVillageForm implements OnInit,AfterViewChecked {
     addVillageForm: FormGroup;
     COORD_REGEXP=/^(-)?[0-9]*$/;
     POPULATION_REGEXP=/^\d*$/;
-    static VILLAGE_REGEXP=/^[A-z]*$/;
+    static VILLAGE_REGEXP=/^\w*$/;
```
Comment (ViacheslavKruhlov) [inline src/main/webapp/components/village/addVillageForm.ts]: I think all these patterns are wrong.

You need to use ' + ' instead ' * '

File: build.gradle (hunk 1 above)
Comment (olleg12) [reply build.gradle]: Didn't notice that. Thnx for being attentive. Fixed!

File: build.gradle (hunk 2 above)
Comment (olleg12) [reply build.gradle]: Fixed

File: npm-debug.log (hunk 3 above)
Comment (olleg12) [reply npm-debug.log]: Fixed, but I'm not sure

File: src/main/java/ua/cv/tim/controller/PlayerController.java (hunk 4 above)
Comment (olleg12) [reply src/main/java/ua/cv/tim/controller/PlayerController.java]: Fixed

File: src/main/webapp/components/village/addVillageForm.ts (hunk 5 above)
Comment (olleg12) [reply src/main/webapp/components/village/addVillageForm.ts]: Fixed

**Review by ViacheslavKruhlov — APPROVED** [review]

**Review by starovoit777 — APPROVED** [review]

**Pull request merged**